staging/src/k8s.io/client-go/tools/cache/controller.go

```diff
@@ -583,6 +583,84 @@ func processDeltas(
 	return nil
 }
 
+// Multiplexes updates in the form of a list of Deltas into a Store, and informs
+// a given handler of events OnUpdate, OnAdd, OnDelete
+func processDeltasInBatch(
+	// Object which receives event notifications from the given deltas
+	handler ResourceEventHandler,
+	clientState Store,
+	deltasList []Deltas,
+	isInInitialList bool,
+) error {
+	// from oldest to newest
+	txns := make([]Transaction, 0)
+	callbacks := make([]func(), 0)
+	txnStore, txnSupported := clientState.(TransactionStore)
+	for _, deltas := range deltasList {
+		for _, d := range deltas {
+			obj := d.Object
+			switch d.Type {
+			case Sync, Replaced, Added, Updated:
+				if old, exists, err := clientState.Get(obj); err == nil && exists {
+					if txnSupported {
+						txns = append(txns, Transaction{
+							OpType: Updated,
+							Object: obj,
+						})
+						callbacks = append(callbacks, func() {
+							handler.OnUpdate(old, obj)
+						})
+					} else {
+						if err := clientState.Update(obj); err != nil {
+							return err
```

Comment:
my quick (potentially incorrect?) skim of the code suggests that an error here results in losing every delta after this one.